manticore/core/smtlib/solver.py

```diff
@@ -352,34 +314,39 @@ def _is_sat(self) -> bool:
 
         :return: whether current state is satisfiable or not.
         """
-        logger.debug("Solver.check() ")
         start = time.time()
-        self._send("(check-sat)")
-        status = self._recv()
+        self._smtlib.send("(check-sat)")
+        status = self._smtlib.recv()
         logger.debug("Check took %s seconds (%s)", time.time() - start, status)
         if status not in ("sat", "unsat", "unknown"):
             raise SolverError(status)
         if consts.defaultunsat:
             if status == "unknown":
                 logger.info("Found an unknown core, probably a solver timeout")
                 status = "unsat"
-
         if status == "unknown":
             raise SolverUnknown(status)
-
-        is_sat = status == "sat"
-        if not is_sat:
-            check_memory_usage()
-            check_disk_usage()
-        return is_sat
+        return status == "sat"
```

Comment:
I thought the `check_memory` and disk usage was to provide more helpful messages? Do we want to get rid of that? @ehennenfent

Reply:
We could guard it with self.debug!
This is global check right? not necessarily related to _this_ solver ?
In linux `dmesg` will show if OOM happened globally too.

Reply:
Strictly speaking those were intended to help diagnose the sporadically-failing SMT failures. We've talked about actually removing them so that we wouldn't need to ship psutil as a dependency. I'm fine with removing them from smtlib here, although if I can find a way to fix them to not need psutil I may leave them in as pre-run checks.

Reply:
I setting this as resolved. It may be better suited to print that stuff when the solver returns "unknown" rather than unsat. Unsat is an expected result in a bunch of places.